docs/source/en/features/gradient_accumulation.md

```diff
@@ -18,20 +15,145 @@ and only update the parameters in the preset iteration.
 
 ## Usage
 
-It is simple to use gradient accumulation in Colossal-AI. Just add this following configuration into your config file.
-The integer represents the number of iterations to accumulate gradients.
+It is simple to use gradient accumulation in Colossal-AI. Just call `booster.no_sync()` which returns a context manager. It accumulate gradients without synchronization meanwhile you should not update the gradients. Add this following configuration into your config file. The integer represents the number of iterations to accumulate gradients.
 
 ```python
 gradient_accumulation = <int>
 ```
 
 ## Hands-on Practice
 
-We provide a [runnable example](https://github.com/hpcaitech/ColossalAI-Examples/tree/main/features/gradient_accumulation)
-to demonstrate gradient accumulation. In this example, we set the gradient accumulation size to be 4. You can run the script using this command:
+We Now demonstrate gradient accumulation. In this example, we set the gradient accumulation size to be 4.
+
+### Step 1. Create a config file
+
+Create a `config.py`.
+```python
+BATCH_SIZE = 128
+NUM_EPOCHS = 200
+
+gradient_accumulation = 4
+```
+
+### Step 2. Import libraries in train.py
+Create a `train.py` and import the necessary dependencies. The version of `torch` should not be lower than 1.8.1.
+
+```python
+import os
+from contextlib import nullcontext
+from pathlib import Path
+
+import torch
+from titans.utils import barrier_context
```

Comment:
You can use https://github.com/hpcaitech/ColossalAI/blob/73d3e4d3091a6f87cd5153194989713dd2429210/colossalai/cluster/dist_coordinator.py#L125